google-cloud-storage/src/main/resources/META-INF/native-image/com/google/cloud/storage/reflect-config.json

```diff
@@ -0,0 +1,27 @@
+[
+  {
+    "name":"org.apache.commons.logging.LogFactory",
+    "allDeclaredFields":true,
+    "allDeclaredMethods":true,
+    "allDeclaredConstructors": true
+  },
+  {
+    "name":"org.apache.commons.logging.impl.Jdk14Logger",
+    "methods":[{"name":"<init>","parameterTypes":["java.lang.String"] }]}
+,
+  {
+    "name":"org.apache.commons.logging.impl.LogFactoryImpl",
+    "allDeclaredFields":true,
+    "allDeclaredMethods":true,
+    "methods":[{"name":"<init>","parameterTypes":[] }]}
+,
```

Comment:
We're not supposed to have any dependency on apache logging. Do you know how I might find the reference path for these classes to try and eliminate them?

Reply:
I looked into this a bit by removing the `LogFactoryImpl` configuration and running `mvn clean install -DskipTests` and `mvn test -Pnative`. This results in the following stacktrace:
```
JUnit Vintage:ITStorageTest:testWriteChannelWithConnectionPool
    MethodSource [className = 'com.google.cloud.storage.it.ITStorageTest', methodName = 'testWriteChannelWithConnectionPool', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       org.apache.http.conn.ssl.SSLConnectionSocketFactory.<clinit>(SSLConnectionSocketFactory.java:151)
       org.apache.http.impl.conn.PoolingHttpClientConnectionManager.getDefaultRegistry(PoolingHttpClientConnectionManager.java:116)
       org.apache.http.impl.conn.PoolingHttpClientConnectionManager.<init>(PoolingHttpClientConnectionManager.java:123)
       com.google.cloud.storage.it.ITStorageTest$CustomHttpTransportFactory.create(ITStorageTest.java:282)
       com.google.cloud.storage.spi.v1.HttpStorageRpc.<init>(HttpStorageRpc.java:110)
       [...]
     Caused by: org.apache.commons.logging.LogConfigurationException: java.lang.ClassNotFoundException: org.apache.commons.logging.impl.LogFactoryImpl (Caused by java.lang.ClassNotFoundException: org.apache.commons.logging.impl.LogFactoryImpl)
       org.apache.commons.logging.LogFactory.createFactory(LogFactory.java:1158)
       org.apache.commons.logging.LogFactory$2.run(LogFactory.java:960)
       java.security.AccessController.doPrivileged(AccessController.java:87)
       org.apache.commons.logging.LogFactory.newFactory(LogFactory.java:957)
       org.apache.commons.logging.LogFactory.getFactory(LogFactory.java:624)
       [...]
     Caused by: java.lang.ClassNotFoundException: org.apache.commons.logging.impl.LogFactoryImpl
       jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:53)
       jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
       java.lang.ClassLoader.loadClass(ClassLoader.java:139)
       org.apache.commons.logging.LogFactory.createFactory(LogFactory.java:1020)
       [...]
```

It looks the usage of LogFactory is coming from `PoolingHttpClientConnectionManager` which is referenced in `ITStorageTest`.

Reply:
Ah okay, so that's testing for an optional feature. Thanks for trying it out